rllib/agents/ppo/ppo.py

```diff
@@ -27,81 +28,160 @@
+        """Initializes a PPOConfig instance.
+
+        # fmt: off
+        # __sphinx_doc_begin__
+
+        Args:
+            lr_schedule: Learning rate schedule. In the format of
+                [[timestep, lr-value], [timestep, lr-value], ...]
+                Intermediary timesteps will be assigned to interpolated learning rate
+                values. A schedule should normally start from timestep 0.
+            use_critic: Should use a critic as a baseline (otherwise don't use value
+                baseline; required for using GAE).
+            use_gae: If true, use the Generalized Advantage Estimator (GAE)
+                with a value function, see https://arxiv.org/pdf/1506.02438.pdf.
+            lambda_: The GAE (lambda) parameter.
+            kl_coeff: Initial coefficient for KL divergence.
+            sgd_minibatch_size: Total SGD batch size across all devices for SGD.
+                This defines the minibatch size within each epoch.
+            num_sgd_iter: Number of SGD iterations in each outer loop (i.e., number of
+                epochs to execute per train batch).
+            shuffle_sequences: Whether to shuffle sequences in the batch when training
+                (recommended).
+            vf_loss_coeff: Coefficient of the value function loss. IMPORTANT: you must
+                tune this if you set vf_share_layers=True inside your model's config.
+            entropy_coeff: Coefficient of the entropy regularizer.
+            entropy_coeff_schedule: Decay schedule for the entropy regularizer.
+            clip_param: PPO clip parameter.
+            vf_clip_param: Clip param for the value function. Note that this is
+                sensitive to the scale of the rewards. If your expected V is large,
+                increase this.
+            grad_clip: If specified, clip the global norm of gradients by this amount.
+            kl_target: Target value for KL divergence.
+
+        # __sphinx_doc_end__
+        # fmt: on
+        """
+
+        super().__init__(trainer_class=PPOTrainer)
+
+        self.lr_schedule = lr_schedule
+        self.use_critic = use_critic
+        self.use_gae = use_gae
+        self.lambda_ = lambda_
+        self.kl_coeff = kl_coeff
+        self.sgd_minibatch_size = sgd_minibatch_size
+        self.num_sgd_iter = num_sgd_iter
+        self.shuffle_sequences = shuffle_sequences
+        self.vf_loss_coeff = vf_loss_coeff
+        self.entropy_coeff = entropy_coeff
+        self.entropy_coeff_schedule = entropy_coeff_schedule
+        self.clip_param = clip_param
+        self.vf_clip_param = vf_clip_param
+        self.grad_clip = grad_clip
+        self.kl_target = kl_target
+
+        # Override some of TrainerConfig's default values with PPO-specific values.
+        self.rollout_fragment_length = 200
+        self.train_batch_size = 4000
+        self.lr = 5e-5
+        self.model["vf_share_layers"] = False
+
+
+# Deprecated: Use ray.rllib.agents.ppo.PPOConfig instead!
```

Comment:
we have a deprecated decorator, right?

Reply:
Correct, which is used here on the __getitem__ method. This way, once a user tries to use the old-style PPO config dict and accesses a key, they get the warning.